drivers/watchdog/wdt_rpi_pico.c

```diff
@@ -154,18 +154,19 @@ static const struct wdt_driver_api wdt_rpi_pico_driver_api = {
 	.feed = wdt_rpi_pico_feed,
 };
 
-#define WDT_RPI_PICO_WDT_DEVICE(idx)                                                               \
-	static const struct wdt_rpi_pico_config wdt_##idx##_config = {                             \
-		.xtal_frequency = DT_INST_PROP_BY_PHANDLE(idx, clocks, clock_frequency)            \
-	};                                                                                         \
-	static struct wdt_rpi_pico_data wdt_##idx##_data = {                                       \
-		.reset_type = WDT_FLAG_RESET_SOC,                                                  \
-		.load = (CONFIG_WDT_RPI_PICO_INITIAL_TIMEOUT *                                     \
-			 RPI_PICO_WDT_TIME_MULTIPLICATION_FACTOR),                                 \
-		.enabled = false                                                                   \
-	};                                                                                         \
-	DEVICE_DT_DEFINE(DT_NODELABEL(wdt##idx), wdt_rpi_pico_init, NULL, &wdt_##idx##_data,       \
-			 &wdt_##idx##_config, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,    \
+#define WDT_RPI_PICO_WDT_DEVICE(idx)                                                             \
+	static const struct wdt_rpi_pico_config wdt_##idx##_config = {                           \
+		.xtal_frequency =                                                                \
+			DT_PROP(DT_CHILD(DT_NODELABEL(syscon), xtal_clk), clock_frequency)       \
```

Comment:
Shouldn't that be `clk_ref`?

Reply:
Yes, I read the docs and it seems to be clk_ref